atlasdb-client/src/main/java/com/palantir/atlasdb/transaction/impl/ConflictDetectionManager.java

```diff
@@ -15,56 +15,133 @@
     }
 
-    public void setConflictDetectionMode(TableReference table, ConflictHandler handler) {
-        overrides.put(table, handler);
-        cachedResult.recompute();
+
+    public static ConflictDetectionManager create(KeyValueService kvs) {
+        ConflictDetectionManager conflictDetectionManager = new ConflictDetectionManager(
+                new CacheLoader<TableReference, ConflictHandler>() {
+                    @Override
+                    public ConflictHandler load(TableReference tableReference) throws Exception {
+                        byte[] metadata = kvs.getMetadataForTable(tableReference);
+                        if (metadata == null) {
+                            return ConflictHandler.RETRY_ON_WRITE_WRITE; // legacy behavior / this is dumb
+                        }
+                        return getConflictHandlerFromMetadata(metadata);
+                    }
+                });
+
+        // kick off an async thread that attempts to fully warm this cache
+        // if it fails (e.g. probably this user has way too many tables), that's okay,
+        // we will be falling back on individually loading in tables as needed.
+        new Thread(() -> {
+            try {
+                conflictDetectionManager.warmCacheWith(
+                        Maps.transformValues(kvs.getMetadataForTables(),
+                                ConflictDetectionManager::getConflictHandlerFromMetadata));
+            } catch (Throwable t) {
+                log.warn("There was a problem with pre-warming the conflict detection cache;"
+                        + " if you have unusually high table scale, this might be expected."
+                        + " Performance may be degraded until normal usage adequately warms the cache.", t);
+            }
+        }, "ConflictDetectionManager Cache Async Pre-Warm").start();
+
+        return conflictDetectionManager;
     }
 
-    public void removeConflictDetectionMode(TableReference table) {
-        overrides.remove(table);
-        cachedResult.recompute();
+    public static ConflictDetectionManager createWithNoConflictDetection() {
+        return new ConflictDetectionManager(
+                new CacheLoader<TableReference, ConflictHandler>() {
+                    @Override
+                    public ConflictHandler load(TableReference tableReference) throws Exception {
+                        return ConflictHandler.IGNORE_ALL;
+                    }
+                });
+    }
+
+    @VisibleForTesting
+    public static ConflictDetectionManager createWithStaticConflictDetection(
+            Map<TableReference, ConflictHandler> staticMap) {
+        CacheLoader<TableReference, ConflictHandler> loader = new CacheLoader<TableReference, ConflictHandler>() {
+            @Override
+            public ConflictHandler load(TableReference tableReference) throws Exception {
+                ConflictHandler conflictHandler = staticMap.get(tableReference);
```

Comment:
I'm hoping to get rid of the RETRY_ON_WRITE_WRITE bullshit once I determine nothing is using this shitty silent fallback behavior